pilot/pkg/grpc/grpc.go

```diff
@@ -0,0 +1,90 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package grpc
+
+import (
+	"context"
+
+	middleware "github.com/grpc-ecosystem/go-grpc-middleware"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/keepalive"
+	"google.golang.org/grpc/status"
+
+	"istio.io/istio/pilot/pkg/features"
+	istiokeepalive "istio.io/istio/pkg/keepalive"
+)
+
+type (
+	SendHandler     func() error
+	ResponseHandler func(err error)
+)
+
+var timeout = features.XdsPushSendTimeout
+
+// Send with timeout if specified. If timeout is zero, sends without timeout.
+func Send(ctx context.Context, send SendHandler, response ResponseHandler) error {
```

Comment:
it looks the response is redundant

Reply:
we can return the err directly and leave to the caller

Reply:
ok. changed